main.tf

```diff
@@ -1,9 +1,10 @@
 resource "aws_customer_gateway" "this" {
   for_each = var.create ? var.customer_gateways : {}
 
-  bgp_asn    = each.value["bgp_asn"]
-  ip_address = each.value["ip_address"]
-  type       = "ipsec.1"
+  bgp_asn     = each.value["bgp_asn"]
+  ip_address  = each.value["ip_address"]
```

Comment:
`ip_address` is optional since [version 4.40](https://github.com/hashicorp/terraform-provider-aws/pull/26673), too. Please change it and update the minimum required version in both `versions.tf`.

Reply:
Looks like `ip_address` is optional only in case if `certificate_arn` is provided, so I've added that as well and made `bgp_asn` optional as well.